pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -198,6 +200,23 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (result re
 		}
 	}
 
+	// re-apply host status from annotation
+
+	content := []byte(annotations[metal3v1alpha1.StatusAnnotation])
+	// Convert the yaml into a typed object
+	obj := &metal3v1alpha1.BareMetalHost{}
+	codecFactory := serializer.NewCodecFactory(r.scheme)
+
+	if err := runtime.DecodeInto(codecFactory.UniversalDecoder(), content, obj); err != nil {
+		reqLogger.Info("failed to decode annotation %v", err)
+	}
+
```

Comment:
this should be probably in the condition. We don't want to do that if we know we will not use the output.